cli/dcos-spark/submit_builder_test.go

```diff
@@ -49,12 +51,89 @@ func (suite *CliTestSuite) TestCleanUpSubmitArgs() {
+	inputArgs := "--conf spark.driver.extraJavaOptions='-XX:+PrintGC -Dparam1=val1 -Dparam2=val2' main.py 100"
+	expected := "--conf=spark.driver.extraJavaOptions=-XX:+PrintGC -Dparam1=val1 -Dparam2=val2"
+	actual, _ := transformSubmitArgs(inputArgs, args.boolVals)
+	assert.Equal(suite.T(), expected, actual[0])
+}
+
+func (suite *CliTestSuite) TestTransformSubmitArgsWithSpecialCharacters() {
+	_, args := sparkSubmitArgSetup()
+	inputArgs := "--conf spark.driver.extraJavaOptions='-Dparam1=\"val 1?\" -Dparam2=val\\ 2! -Dmulti.dot.param3=\"val\\ 3\" -Dpath=$PATH' main.py 100"
+	expected := "--conf=spark.driver.extraJavaOptions=-Dparam1=\"val 1?\" -Dparam2=val\\ 2! -Dmulti.dot.param3=\"val\\ 3\" -Dpath=$PATH"
+	actual, _ := transformSubmitArgs(inputArgs, args.boolVals)
+	assert.Equal(suite.T(), expected, actual[0])
+}
+
+func (suite *CliTestSuite) TestTransformSubmitArgsConfsAlreadyHasEquals() {
+	_, args := sparkSubmitArgSetup()
+	inputArgs := "--conf=spark.driver.extraJavaOptions='-Dparam1=val1' main.py 100"
+	expected := "--conf=spark.driver.extraJavaOptions=-Dparam1=val1"
+	actual, _ := transformSubmitArgs(inputArgs, args.boolVals)
+	assert.Equal(suite.T(), expected, actual[0])
+}
+
+func (suite *CliTestSuite) TestTransformSubmitArgsMultilines() {
+	_, args := sparkSubmitArgSetup()
+	inputArgs := `--conf spark.driver.extraJavaOptions='-XX:+PrintGC -XX:+PrintGCTimeStamps' \
+	--supervise --driver-memory 1g \
+	main.py 100`
+	expected := []string{"--conf=spark.driver.extraJavaOptions=-XX:+PrintGC -XX:+PrintGCTimeStamps", "--supervise", "--driver-memory=1g", "main.py"}
+	actual, _ := transformSubmitArgs(inputArgs, args.boolVals)
+	assert.Equal(suite.T(), expected, actual)
+}
+
+func (suite *CliTestSuite) TestIsSparkApp() {
+	assert.True(suite.T(), isSparkApp("mainApp.jar"))
+	assert.True(suite.T(), isSparkApp("pythonFile.py"))
+	assert.True(suite.T(), isSparkApp("file.R"))
+	assert.False(suite.T(), isSparkApp("app.c"))
+	assert.False(suite.T(), isSparkApp("randomFlag"))
+}
+
+func (suite *CliTestSuite) TestCheckIfBooleanFlag() {
+	_, args := sparkSubmitArgSetup()
+	idx := 0
+
+	inputArgs := []string{"--supervise", "--conf", "spark.mesos.containerizer=mesos"}
+	isBool, skipNext := checkIfBooleanFlag(args.boolVals, inputArgs, idx)
+	assert.True(suite.T(), isBool, "isBool should be true since --supervise is a Boolean flag")
+	assert.False(suite.T(), skipNext, "--conf is not a Boolean value")
+}
+
+func (suite *CliTestSuite) TestCheckIfBooleanFlagWithBooleanValue() {
+	_, args := sparkSubmitArgSetup()
+	idx := 0
+
+	inputArgs := []string{"--supervise", "true", "--conf", "spark.mesos.containerizer=mesos"}
+	isBool, skipNext := checkIfBooleanFlag(args.boolVals, inputArgs, idx)
+	assert.True(suite.T(), isBool, "isBool should be true since --supervise is a Boolean flag")
+	assert.True(suite.T(), skipNext, "skipNext should be true since 'true' is a Boolean value")
+
+	inputArgs = []string{"--supervise", "false", "--conf", "spark.mesos.containerizer=mesos"}
```

Comment:
What about `--supervise` with NO args?

I also don't see a test for the equivalence of:
* `--supervise` and `--supervise true`
* (missing `--supervise`) and `--supervise false`

Reply:
I'm all for not supporting `--bool-flag BOOL_VAL` options, but then we need to change the handling of the bool args. The thing to confirm would then be that we correctly handle the different cases:
* `--supervise` should pass
* `--supervise true` should error
* `--supervise false` should error
* `--supervise https://some.jar` should pass

Are these all the test cases that we need to cover?